sopel/loader.py

```diff
@@ -234,15 +234,16 @@ def clean_module(module, config):
     urls = []
     for obj in itervalues(vars(module)):
         if callable(obj):
+            is_sopel_callable = getattr(obj, '_sopel_callable', False) is True
             if getattr(obj, '__name__', None) == 'shutdown':
                 shutdowns.append(obj)
-            elif is_triggerable(obj):
+            elif is_sopel_callable and is_triggerable(obj):
                 clean_callable(obj, config)
                 callables.append(obj)
-            elif hasattr(obj, 'interval'):
+            elif is_sopel_callable and hasattr(obj, 'interval'):
                 clean_callable(obj, config)
                 jobs.append(obj)
-            elif is_url_callback(obj):
+            elif is_sopel_callable and is_url_callback(obj):
```

Comment:
Was there a reason you didn't go with a well-placed `continue` instead of adding `is_sopel_callable and` to every `if`?

```python
    for obj in itervalues(vars(module)):
        if callable(obj):
            is_sopel_callable = getattr(obj, '_sopel_callable', False) is True
            if getattr(obj, '__name__', None == 'shutdown':
                shutdowns.append(obj)
            elif not is_sopel_callable:
                continue  # skip it
            elif is_triggerable(obj):
                # et cetera; I'm not gonna retype everything xD
```

(GitHub won't let me make an easily-applied suggestion because the range includes deleted lines. Sigh.)

I get that `setup` (irrelevant here) and `shutdown` won't be decorated with something that adds the `_sopel_callable` attribute, but a `continue` after checking for the special cases should be fine and avoids re-evaluating the same condition over and over.

Reply:
Yeah you're right. Why I did that is beyond me.